PracticeProject/API/API.swift

```diff
@@ -5,26 +5,17 @@
 
 import Foundation
 
+/// It is API protocol used for common Api handling
 protocol API {
-    init()
-
-    func fetchUsers(excludingUserWithID: String?, success: (UsersList) -> Void, failure: (FetchError) -> Void)
+    
+    /// Getting the response from user API's
+    /// - Parameters:
+    ///   - excludingUserWithID: It is exclude the user from user list
+    ///   - success: get the user list response
+    ///   - failure: get the error message
+    func fetchUsers(excludingUserWithID: String?, success: @escaping (UsersList?) -> Void, failure: @escaping (FetchError?) -> Void)
 }
 
-extension API {
```

Comment:
The `API` static extension method `make(/*can add arguments*/)` is intended to hide *who (what object)* implements the `API` to the rest of the application. Therefore, if a new `API` implementation was to replace the current one, returning the new object from this function will be the only thing required, i.e. no other change in the app. 
Can you make some changes to achieve this goal? Or ask if this is not clear @iosgnanavel

Reply:
I don’t have any idea to use the API make method. Kindly share any document regarding this. Really, it'll help to implement it properly.